src/Uno.UWP/ApplicationModel/Chat/ChatMessageReader.Android.cs

```diff
@@ -0,0 +1,120 @@
+﻿#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Chat
+{
+	public partial class ChatMessageReader
+	{
+		// https://stackoverflow.com/questions/848728/how-can-i-read-sms-messages-from-the-device-programmatically-in-android
+
+		private TimeSpan _range;
+		private Android.Database.ICursor _cursor = null;
+		private int _colBody, _colTime, _colInOut, _colRead, _colFrom, _colTo, _colSeen, _colAddr, _colStatus;
+
+		internal ChatMessageReader(TimeSpan range)
+		{
+			_range = range;
+			Android.Content.ContentResolver cr = Android.App.Application.Context.ContentResolver;
+			//Android.Net.Uri messageUri = Android.Net.Uri.Parse("content://sms/");
+
+			_cursor = cr.Query(Android.Provider.Telephony.Sms.ContentUri, null, null, null, Android.Provider.Telephony.Sms.DefaultSortOrder);
+
+			if (_cursor.MoveToFirst())
+			{ // runtime optimizations
+				_colBody = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Body);
+				_colTime = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Date); //int
+				_colInOut = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Type); // 1:inbox, 4:outbox, 2:sent, ...
+				_colRead = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Read); // int (bool)
+				_colAddr = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Address);
+				_colSeen = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Seen); // int (bool)
+				_colStatus = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Status); // int (bool)
+																											//0: _id
+																											//1: thread_id
+																											//2: address
+																											//3: person
+																											//4: date
+																											//5: protocol
+																											//6: read
+																											//7: status
+																											//8: type
+																											//9: reply_path_present
+																											//10: subject
+																											//11: body
+																											//12: service_center
+																											//13: locked
+			}
+		}
+		~ChatMessageReader()
+		{
+			if (_cursor != null)
+			{
+				_cursor.Close();
+			}
+		}
+
+		public IAsyncOperation<IReadOnlyList<Windows.ApplicationModel.Chat.ChatMessage>> ReadBatchAsync()
+			=> ReadBatchAsyncTask().AsAsyncOperation<IReadOnlyList<Windows.ApplicationModel.Chat.ChatMessage>>();
+
+		public async Task<IReadOnlyList<Windows.ApplicationModel.Chat.ChatMessage>> ReadBatchAsyncTask()
+		{
+			var entriesList = new List<Windows.ApplicationModel.Chat.ChatMessage>();
+
+			if (_cursor is null || _cursor.IsAfterLast)
+			{
+				return entriesList;
+			}
+
+			for (int pageGuard = 100; pageGuard > 0; pageGuard--)

```

Comment:
The `100` should come from a default value provided by `ReadBatchAsync(int)`.

Reply:
No, as `int` in `ReadBatchAsync(int)` is about count of messages; and  `ReadBatchAsync()` returns batch of messages, not all of them, and (as far as I know) not always same number of messages. `pageGuard` guards against "eternal loop", and limits "UWP batch size" to 100 messages.
As my app uses `ReadBatchAsync()`, I implemented this method, not `ReadBatchAsync(int)`.